### PR TITLE
chore: replace panic! with internal_error! in compiler modules

### PR DESCRIPTION
**Pull request by jarlah**
relates to https://github.com/roc-lang/roc/issues/2046

### CODE REVIEW

**Comment by jarlah** [comment]
Was unable to run clippy since i have m2 MacBook. So if Workflow runs is approved they might fail. If so i will fix them.

**Comment by jarlah** [comment]
fixed compilation issue in llvm.rs

**Comment by jarlah** [comment]
I seem to have introduced a bug in test_gen. I will investigate. Possibly do the art of elimination.

File: crates/compiler/test_gen/src/helpers/llvm.rs [hunk 1]
```diff
@@ -611,8 +612,8 @@ where
             std::mem::forget(given);
         }
         Err((msg, tag)) => match tag {
-            CrashTag::Roc => panic!(r#"Roc failed with message: "{}""#, msg),
-            CrashTag::User => panic!(r#"User crash with message: "{}""#, msg),
```
Comment (ayazhafiz) [inline crates/compiler/test_gen/src/helpers/llvm.rs:615]: I think the test failures may be due to changing these lines. The way our internal tests work, I believe they require just a standard panic here.

File: crates/compiler/test_gen/src/helpers/wasm.rs [hunk 2]
```diff
@@ -207,11 +208,12 @@ impl<'a> ImportDispatcher for TestDispatcher<'a> {
                 1 => format!(r#"User crash with message: "{}""#, roc_msg),
                 tag => format!(r#"Got an invald panic tag: "{}""#, tag),
             };
-            panic!("{}", msg)
```
Comment (ayazhafiz) [inline crates/compiler/test_gen/src/helpers/wasm.rs:210]: I think the test failures may be due to changing these lines. The way our internal tests work, I believe they require just a standard panic here.

File: crates/compiler/test_gen/src/helpers/wasm.rs [hunk 3]
```diff
@@ -376,7 +378,7 @@ macro_rules! assert_evals_to {
         let phantom = std::marker::PhantomData;
         let _ = $ignore_problems; // Always ignore "problems"! One backend (LLVM) is enough to cover them.
         match $crate::helpers::wasm::assert_evals_to_help::<$ty>($src, phantom) {
-            Err(msg) => panic!("{}", msg),
```
Comment (ayazhafiz) [inline crates/compiler/test_gen/src/helpers/wasm.rs:379]: Same comment as above

File: crates/compiler/test_gen/src/helpers/wasm.rs [hunk 4]
```diff
@@ -400,7 +402,7 @@ macro_rules! assert_refcounts {
         let result =
             $crate::helpers::wasm::assert_wasm_refcounts_help::<$ty>($src, phantom, num_refcounts);
         match result {
-            Err(msg) => panic!("{:?}", msg),
```
Comment (ayazhafiz) [inline crates/compiler/test_gen/src/helpers/wasm.rs:403]: Same comment as above

**Comment by jarlah** [comment]
cool. Thanks @ayazhafiz I will take a look :)

**Comment by jarlah** [comment]
first thing I synced with upstream. So this force push was just updating

**Comment by jarlah** [comment]
ill revert the changes to the test_gen package later.

File: crates/compiler/test_gen/src/helpers/wasm.rs (hunk 4 above)
Comment (jarlah) [reply crates/compiler/test_gen/src/helpers/wasm.rs]: fixed in https://github.com/roc-lang/roc/compare/100a2df04327f19680e578c0f1bb2b87355b4311..e5eb5717eace1559d73561099e9381630b497634

File: crates/compiler/test_gen/src/helpers/wasm.rs (hunk 3 above)
Comment (jarlah) [reply crates/compiler/test_gen/src/helpers/wasm.rs]: fixed in https://github.com/roc-lang/roc/compare/100a2df04327f19680e578c0f1bb2b87355b4311..e5eb5717eace1559d73561099e9381630b497634

File: crates/compiler/test_gen/src/helpers/wasm.rs (hunk 2 above)
Comment (jarlah) [reply crates/compiler/test_gen/src/helpers/wasm.rs]: fixed in https://github.com/roc-lang/roc/compare/100a2df04327f19680e578c0f1bb2b87355b4311..e5eb5717eace1559d73561099e9381630b497634

File: crates/compiler/test_gen/src/helpers/llvm.rs (hunk 1 above)
Comment (jarlah) [reply crates/compiler/test_gen/src/helpers/llvm.rs]: fixed in

**Review by ayazhafiz — APPROVED** [review]
Great, thanks so much!

**Pull request merged**